lib/modules/datasource/docker/index.ts

```diff
@@ -812,54 +812,144 @@ export class DockerDatasource extends Datasource {
   @cache({
     namespace: 'datasource-docker-digest',
     key: (
-      { registryUrl, packageName }: GetReleasesConfig,
+      { registryUrl, packageName, currentDigest }: DigestConfig,
       newValue?: string
     ) => {
       const newTag = newValue ?? 'latest';
       const { registryHost, dockerRepository } = getRegistryRepository(
         packageName,
         registryUrl!
       );
-      return `${registryHost}:${dockerRepository}:${newTag}`;
+      const digest = currentDigest ? '@' + currentDigest : '';
+      return `${registryHost}:${dockerRepository}:${newTag}${digest}`;
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
-      let manifestResponse = await this.getManifestResponse(
-        registryHost,
-        dockerRepository,
-        newTag,
-        'head'
-      );
-      if (manifestResponse) {
-        if (hasKey('docker-content-digest', manifestResponse.headers)) {
-          digest =
-            (manifestResponse.headers['docker-content-digest'] as string) ||
-            null;
-        } else {
-          logger.debug(
-            { registryHost },
-            'Missing docker content digest header, pulling full manifest'
-          );
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
+        let manifestResponse = await this.getManifestResponse(
+          registryHost,
+          dockerRepository,
+          currentDigest,
+          'head'
+        );
+
+        if (
+          manifestResponse &&
+          (manifestResponse.headers['content-type'] === MediaType.manifestV2 ||
+            manifestResponse.headers['content-type'] ===
```

Comment:
```suggestion
          (manifestResponse?.headers['content-type'] === MediaType.manifestV2 ||
            manifestResponse?.headers['content-type'] ===
```
simplify

Reply:
Fixed in https://github.com/renovatebot/renovate/pull/16554/commits/65fc2280dd9a23c17551a71e0c482aaabbe8a9d1